File: Data Structures & Algorithms/last-stone-weight/submission-0.py
import heapq
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        if len(stones) == 1:
            return stones[0]
        
        heap = []
        for sw in stones:
            heapq.heappush(heap, -sw)

        while len(heap) > 1:
            sw1 = -heapq.heappop(heap)
            sw2 = -heapq.heappop(heap)
            diff = abs(sw1-sw2)
            if diff != 0:
                heapq.heappush(heap, -diff)

        return -heap[0] if len(heap) == 1 else 0

File: Data Structures & Algorithms/pow-x-n/submission-0.java
class Solution {
    public double myPow(double x, int n) {
        if(n == 0)
            return 1.0;
        
        double total = 1.0;
        for(int i = 1; i <=Math.abs(n); ++i){
            if(n < 0)
                total *= 1/x;  
            else 
                total *= x;  
        }
        return total;
    }
}
